extensions/data-transfer/portability-data-transfer-generic/src/main/java/org/datatransferproject/datatransfer/generic/GenericTransferExtension.java

```diff
@@ -0,0 +1,214 @@
+    JobStore jobStore = context.getService(JobStore.class);
+    TransferServiceConfig configuration = context.getService(TransferServiceConfig.class);
+    if (configuration.getServiceConfig().isEmpty()) {
+      throw new RuntimeException("Empty service configuration");
+    }
+    GenericTransferServiceConfig serviceConfig;
+    try {
+      serviceConfig = parseConfig(configuration.getServiceConfig().get());
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException("Invalid service configuration", e);
+    }
+
+    AppCredentialStore appCredentialStore = context.getService(AppCredentialStore.class);
+    String serviceNameUpper = serviceConfig.getServiceId().toUpperCase();
+    AppCredentials appCredentials;
+    try {
+      appCredentials =
+          appCredentialStore.getAppCredentials(
+              format("%s_KEY", serviceNameUpper), format("%s_SECRET", serviceNameUpper));
+    } catch (IOException e) {
+      throw new RuntimeException(
+          format(
+              "Failed to get application credentials for %s (%s)",
+              serviceNameUpper, serviceConfig.getServiceId()),
+          e);
+    }
+
+    if (serviceConfig.supportsVertical(BLOBS)) {
+      importerMap.put(
+          BLOBS,
+          new GenericFileImporter<BlobbyStorageContainerResource, BlobbySerializer.ExportData>(
+              BlobbySerializer::serialize,
+              appCredentials,
+              urlAppend(serviceConfig.getEndpoint(), "blobs"),
+              jobStore,
+              context.getMonitor()));
+    }
+
+    if (serviceConfig.supportsVertical(MEDIA)
+        // PHOTOS and VIDEOS can be mapped from MEDIA
+        || serviceConfig.supportsVertical(PHOTOS)
+        || serviceConfig.supportsVertical(VIDEOS)) {
+      importerMap.put(
+          MEDIA,
+          new GenericFileImporter<MediaContainerResource, MediaSerializer.ExportData>(
+              MediaSerializer::serialize,
+              appCredentials,
+              urlAppend(serviceConfig.getEndpoint(), "media"),
+              jobStore,
+              context.getMonitor()));
+    }
+
+    if (serviceConfig.supportsVertical(SOCIAL_POSTS)) {
+      importerMap.put(
+          SOCIAL_POSTS,
+          new GenericImporter<SocialActivityContainerResource, SocialPostsSerializer.ExportData>(
+              SocialPostsSerializer::serialize,
+              appCredentials,
+              urlAppend(serviceConfig.getEndpoint(), "social-posts"),
+              context.getMonitor()));
+    }
+
+    if (serviceConfig.supportsVertical(CALENDAR)) {
+      importerMap.put(
+          CALENDAR,
+          new GenericImporter<CalendarContainerResource, CalendarSerializer.ExportData>(
+              CalendarSerializer::serialize,
+              appCredentials,
+              urlAppend(serviceConfig.getEndpoint(), "calendar"),
+              context.getMonitor()));
+    }
+  }
+
+  private URL urlAppend(URL base, String suffix) {
```

Comment:
Sure is! 🙂 
I think because most other extensions use SDKs they don't have to deal with URLs manually, but definitely could be moved to a utility class if we spot it being reimplemented elsewhere